lib/puppet/pops/types/p_runtime_type.rb

```diff
@@ -0,0 +1,90 @@
+module Puppet::Pops
+module Types
+
+# @api public
+class PRuntimeType < PAnyType
+  TYPE_NAME_OR_PATTERN = PVariantType.new([PStringType::NON_EMPTY, PTupleType.new([PRegexpType::DEFAULT, PStringType::NON_EMPTY])])
+
+  attr_reader :runtime, :name_or_pattern
+
+  # Creates a new instance of a Runtime type
+  #
+  # @param runtime [String] the name of the runtime, e.g. 'ruby'
+  # @param name_or_pattern [String,Array(Regexp,String)] name of runtime or two patterns, mapping Puppet name => runtime name
+  # @api public
+  def initialize(runtime, name_or_pattern)
+    unless runtime.nil? || runtime.is_a?(Symbol)
+      runtime = TypeAsserter.assert_instance_of("Runtime 'runtime'", PStringType::NON_EMPTY, runtime).to_sym
+    end
+    @runtime = runtime
+    @name_or_pattern = TypeAsserter.assert_instance_of("Runtime 'name_or_pattern'", TYPE_NAME_OR_PATTERN, name_or_pattern, true)
+  end
+
+  def hash
+    @runtime.hash ^ @name_or_pattern.hash
+  end
+
+  def eql?(o)
+    self.class == o.class && @runtime == o.runtime && @name_or_pattern == o.name_or_pattern
+  end
+
+  def instance?(o, guard = nil)
+    assignable?(TypeCalculator.infer(o), guard)
+  end
+
+  def iterable?(guard = nil)
+    if @runtime == :ruby && !runtime_type_name.nil?
+      begin
+        c = ClassLoader.provide(self)
+        return c < Iterable unless c.nil?
+      rescue ArgumentError
+      end
+    end
+    false
+  end
+
+  def iterable_type(guard = nil)
+    iterable?(guard) ? PIterableType.new(self) : nil
+  end
+
+  def runtime_type_name
```

Comment:
It was public previously (attr_reader). Not sure what the rules are for marking it @api private